sdk/src/main/java/io/radar/sdk/model/RadarUser.kt

```diff
@@ -143,9 +155,21 @@ class RadarUser(
             val dma = obj.optJSONObject(FIELD_DMA)?.let(RadarRegion.Companion::fromJson)
             val postalCode = obj.optJSONObject(FIELD_POSTAL_CODE)?.let(RadarRegion.Companion::fromJson)
 
-            val nearbyChains = obj.optJSONArray(FIELD_NEARBY_PLACE_CHAINS)?.let { nearbyArr ->
-                (0 until nearbyArr.length()).map { idx ->
-                    RadarChain.fromJson(nearbyArr.optJSONObject(idx) ?: JSONObject())
+            val nearbyChains = obj.optJSONArray(FIELD_NEARBY_PLACE_CHAINS)?.let { nearbyChainsArr ->
+                (0 until nearbyChainsArr.length()).map { idx ->
+                    RadarChain.fromJson(nearbyChainsArr.optJSONObject(idx) ?: JSONObject())
+                }
+            }?.toTypedArray()
+
+            val segments = obj.optJSONArray(FIELD_SEGMENTS)?.let { segmentsArr ->
+                (0 until segmentsArr.length()).map { idx ->
+                    RadarSegment.fromJson(segmentsArr.optJSONObject(idx) ?: JSONObject())
+                }
+            }?.toTypedArray()
+
+            val topChains = obj.optJSONArray(FIELD_TOP_CHAINS)?.let { topChainsArr ->
```

Comment:
nit: future note, we should probably create some generic out of this that can take a `FIELD_*` and a class name that abides by some interface that includes `fromJson` and `fromJson` to deserialize

Reply:
Totally, this should all be refactored at some point